tests/pyfunc/test_pyfunc_schema_enforcement.py

```diff
@@ -2077,6 +2077,44 @@ def predict(self, context, model_input, params=None):
     np.testing.assert_equal(result, expected_df.values.tolist()[0])
 
 
+def test_input_example_validation_during_logging(tmp_path):
+    class MyModel(mlflow.pyfunc.PythonModel):
+        def predict(self, context, model_input, params=None):
+            assert isinstance(model_input, pd.DataFrame)
+            return "string"
+
+    with pytest.raises(MlflowException, match="Failed to validate serving input example."):
```

Comment:
```suggestion
    with pytest.raises(MlflowException, match=r"Failed to validate serving input example\."):
```

feel free to ignore this. just my habit for strictness.